File: src/main/java/pl/put/poznan/sorting/models/SortResponse.java
package pl.put.poznan.sorting.models;

import java.util.Map;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

public class SortResponse {

    public SortResponse(Object[] data, Map<String, Integer> time) {
        this.data = data;
        this.time = time;
    }

    public final Object[] data;
    public final Map<String, Integer> time;

    public static SortResponse fromJson(String json) throws JsonSyntaxException {
        return new Gson().fromJson(json, SortResponse.class);
    }

    public String toJson() {
        return new Gson().toJson(this);
    }

}

File: lib/sort_api/models/sort_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'sort_response.g.dart';

@JsonSerializable()
class SortResponse {
  SortResponse({
    required this.algorithm,
    required this.data,
    required this.time,
  });

  factory SortResponse.fromJson(Map<String, dynamic> json) =>
      _$SortResponseFromJson(json);

  final String algorithm;
  final List<dynamic> data;
  final String time;

  Map<String, dynamic> toJson() => _$SortResponseToJson(this);
}
